File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if(root==None):
            return 0
        if(root.left==None and root.right==None):
            return 1
        
        return max(self.calculateHeight(root.left, 1), self.calculateHeight(root.right, 1))
    
    def calculateHeight(self, r, cnt):
            if(r==None):
                return cnt
            
            return max(self.calculateHeight(r.left, cnt+1), self.calculateHeight(r.right, cnt+1))

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        if(root==null){
            return 0;
        }
        if(root.left==null && root.right==null){
            return 1;
        }
        return Math.max(calculateHeight(root.left, 1), calculateHeight(root.right, 1));
    }
    
    public int calculateHeight(TreeNode r, int cnt){
        if(r==null){
            return cnt;
        }
        return Math.max(calculateHeight(r.left, cnt+1), calculateHeight(r.right, cnt+1));
    }
}